openmetadata-ui/src/main/resources/ui/src/components/Explore/ExploreTree/ExploreTree.tsx

```diff
@@ -0,0 +1,183 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { Tree, Typography } from 'antd';
+import classNames from 'classnames';
+import { uniqueId } from 'lodash';
+import React, { useCallback, useState } from 'react';
+import { EntityFields } from '../../../enums/AdvancedSearch.enum';
+import { SearchIndex } from '../../../enums/search.enum';
+import { getAggregateFieldOptions } from '../../../rest/miscAPI';
+import { getCountBadge } from '../../../utils/CommonUtils';
+import { getEntityNameLabel } from '../../../utils/EntityUtils';
+import {
+  getAggregations,
+  getSubLevelHierarchyKey,
+  updateTreeData,
+} from '../../../utils/Explore.utils';
+import searchClassBase from '../../../utils/SearchClassBase';
+import serviceUtilClassBase from '../../../utils/ServiceUtilClassBase';
+import { getEntityIcon } from '../../../utils/TableUtils';
+import {
+  ExploreTreeNode,
+  ExploreTreeProps,
+  TreeNodeData,
+} from './ExploreTree.interface';
+
+const ExploreTreeTitle = ({ node }: { node: ExploreTreeNode }) => (
+  <Typography.Text
+    className={classNames({
+      'm-l-xs': node.data?.isRoot,
+    })}
+    data-testid={`explore-tree-title-${node.title}`}>
+    {node.title}
+  </Typography.Text>
+);
+
+const ExploreTree = ({ onFieldValueSelect }: ExploreTreeProps) => {
+  const initTreeData = searchClassBase.getExploreTree();
+  const [treeData, setTreeData] = useState(initTreeData);
+
+  const onLoadData = useCallback(
+    async (treeNode: ExploreTreeNode) => {
+      if (treeNode.children) {
+        return;
+      }
+
+      const {
+        isRoot = false,
+        currentBucketKey,
+        currentBucketValue,
+        filterField = [],
+        rootIndex,
+      } = treeNode?.data as TreeNodeData;
+
+      const searchIndex = isRoot
+        ? treeNode.key
+        : treeNode?.data?.parentSearchIndex;
+
+      const { bucket: bucketToFind, queryFilter } = getSubLevelHierarchyKey(
+        rootIndex === SearchIndex.DATABASE,
+        currentBucketKey as EntityFields,
+        currentBucketValue
+      );
+
+      const res = await getAggregateFieldOptions(
+        searchIndex as SearchIndex,
+        bucketToFind,
+        '',
+        JSON.stringify(queryFilter)
+      );
+      const aggregations = getAggregations(res.data.aggregations);
+      const buckets = aggregations[bucketToFind].buckets;
+      const isServiceType = bucketToFind === EntityFields.SERVICE_TYPE;
+      const isEntityType = bucketToFind === EntityFields.ENTITY_TYPE;
+
+      const sortedBuckets = [...buckets].sort((a, b) =>
```

Comment:
if buckets is already an array, why do we need to recreate the array and spread it again?